### PR TITLE
Upgrade Travis CMake version

### PR DESCRIPTION
**Pull request by klshrinidhi**
This is related to PR #925 which bumps up the CMake version required on Linux and OSX to be the same as that for Windows.

### CODE REVIEW

**Comment by klshrinidhi** [comment]
:question: :grey_question: :question: :grey_question: :question: :grey_question:

**Comment by chrisdembia** [comment]
See comments in #925

File: .travis.yml [hunk 1]
```diff
@@ -73,6 +73,10 @@ before_install:
   # (e.g., literally a \t in a source file).
   - if grep --line-num --recursive --exclude-dir="*dependencies*" --include={CMakeLists.txt,*.cpp,*.c,*.h} -P "\t" . ; then echo "Tabs found in the lines shown above. See CONTRIBUTING.md about tabs."; false; fi
 
+  - if [[ "$TRAVIS_OS_NAME" = "osx" ]]; then brew update; fi
+  - if [[ "$TRAVIS_OS_NAME" = "osx" ]]; then brew unlink cmake; fi
+  - if [[ "$TRAVIS_OS_NAME" = "osx" ]]; then brew install cmake; fi
+  - cmake --version
```
Comment (chrisdembia) [inline .travis.yml]: Can you add a space under this line?

File: .travis.yml [hunk 2]
```diff
@@ -168,7 +172,7 @@ script:
 
   ## Test python wrapping.
   # Add OpenSim libraries to library path.
-  - if [ "$WRAP" = "on" ]; then export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$OPENSIM_HOME/lib/x86_64-linux-gnu:$OPENSIM_DEPENDENCIES_INSTALL_DIR/BTK/lib/btk-0.4dev; fi
+  - if [ "$WRAP" = "on" ]; then export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$OPENSIM_HOME/lib:$OPENSIM_HOME/lib/x86_64-linux-gnu:$OPENSIM_DEPENDENCIES_INSTALL_DIR/BTK/lib/btk-0.4dev; fi
```
Comment (chrisdembia) [inline .travis.yml]: Why was this necessary?

**Comment by chrisdembia** [comment]
Perhaps wait to address the feedback until after the dev meeting.

File: .travis.yml (hunk 2 above)
Comment (klshrinidhi) [inline .travis.yml]: The location of the libraries changed. Not sure why.

Comment (chrisdembia) [inline .travis.yml]: Ahh here's the reason: https://github.com/Kitware/CMake/commit/620939e4e6f5a61cd5c0fac2704de4bfda0eb7ef

Above, we could remove `x86_64-linux-gnu` from LD_LIBRARY_PATH. 
I wonder if there are some other changes we need to make related to this...I glanced at the README and didn't find anything that would need to change.
